src/index.tsx

```diff
@@ -46,26 +56,154 @@ function Updaters() {
   )
 }
 
+const Web3Wrapper = () => {
+  const dispatch = useAppDispatch()
+  const { hooks } = useWeb3React()
+  const walletOverrideBackfilled = useAppSelector((state) => state.user.walletOverrideBackfilled)
+
+  const injectedIsActive = hooks.useSelectedIsActive(injected)
+  const coinbaseWalletIsActive = hooks.useSelectedIsActive(coinbaseWallet)
+  const walletConnectIsActive = hooks.useSelectedIsActive(walletConnect)
+
+  const previousInjectedIsActive = usePrevious(injectedIsActive)
+  const previousCoinbaseWalletIsActive = usePrevious(coinbaseWalletIsActive)
+  const previousWalletConnectIsActive = usePrevious(walletConnectIsActive)
+
+  const injectedIsActivating = hooks.useSelectedIsActivating(injected)
+  const coinbaseWalletIsActivating = hooks.useSelectedIsActivating(coinbaseWallet)
+  const walletConnectIsActivating = hooks.useSelectedIsActivating(walletConnect)
+
+  const previousInjectedIsActivating = usePrevious(injectedIsActivating)
+  const previousCoinbaseWalletIsActivating = usePrevious(coinbaseWalletIsActivating)
+  const previousWalletConnectIsActivating = usePrevious(walletConnectIsActivating)
+
+  const [isInjectedEagerlyConnecting, setIsInjectedEagerlyConnecting] = useReducer(
```

Comment:
I think these would be simpler as `useState`s, because there are no side effects or computations.